src/test/java/educationalproject/programmingstuff/tests_units/UserServiceImplTest.java

```diff
@@ -0,0 +1,94 @@
+package educationalproject.programmingstuff.tests_units;
+
+import educationalproject.programmingstuff.TestDataCreator;
+import educationalproject.programmingstuff.model.User;
+import educationalproject.programmingstuff.repositories.UserRepository;
+import educationalproject.programmingstuff.servicies.UserServiceImpl;
+import educationalproject.programmingstuff.servicies.dto.UserCreateRequestDto;
+import educationalproject.programmingstuff.servicies.dto.UserResponseDto;
+import educationalproject.programmingstuff.servicies.mappers.UserCreateRequestMapperImpl;
+import educationalproject.programmingstuff.servicies.mappers.UserResponseMapperImpl;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.times;
+
+@ExtendWith(SpringExtension.class)
+class UserServiceImplTest {
+
+    @Mock
+    UserRepository userRepository;
+
+    @Spy
+    UserResponseMapperImpl userResponseMapper; // todo q: Can't spy interface. Even dude on stackoverflow can't
```

Comment:
because it should be like this:

@Spy
private ClientMapper clientMapper = Mappers.getMapper(ClientMapper.class);